src/model/AirLoopHVACUnitaryHeatPumpAirToAir.cpp

```diff
@@ -388,28 +392,41 @@ namespace model {
           // ! hvacComponent.optionalCast<FanSystemModel>()
       ) {
         return false;
-      };
+      }
 
       return setPointer(OS_AirLoopHVAC_UnitaryHeatPump_AirToAirFields::SupplyAirFanName, hvacComponent.handle());
-      ;
     }
 
     bool AirLoopHVACUnitaryHeatPumpAirToAir_Impl::setHeatingCoil(HVACComponent& hvacComponent) {
-      if (!hvacComponent.optionalCast<CoilHeatingDXSingleSpeed>()) {
-        return false;
-      };
+      bool isTypeOK = false;
 
-      return setPointer(OS_AirLoopHVAC_UnitaryHeatPump_AirToAirFields::HeatingCoilName, hvacComponent.handle());
-      ;
+      if (hvacComponent.optionalCast<CoilHeatingDXSingleSpeed>()) {
+        isTypeOK = true;
+      } else if (hvacComponent.optionalCast<CoilHeatingDXVariableSpeed>()) {
+        isTypeOK = true;
+      }
+
+      if (isTypeOK) {
+        return setPointer(OS_AirLoopHVAC_UnitaryHeatPump_AirToAirFields::HeatingCoilName, hvacComponent.handle());
+      }
+      return false;
     }
 
     bool AirLoopHVACUnitaryHeatPumpAirToAir_Impl::setCoolingCoil(HVACComponent& hvacComponent) {
-      if (!(hvacComponent.optionalCast<CoilCoolingDXSingleSpeed>() || hvacComponent.optionalCast<CoilSystemCoolingDXHeatExchangerAssisted>())) {
-        return false;
-      };
+      bool isTypeOK = false;
 
-      return setPointer(OS_AirLoopHVAC_UnitaryHeatPump_AirToAirFields::CoolingCoilName, hvacComponent.handle());
-      ;
+      if (hvacComponent.optionalCast<CoilCoolingDXSingleSpeed>()) {
+        isTypeOK = true;
+      } else if (hvacComponent.optionalCast<CoilCoolingDXVariableSpeed>()) {
+        isTypeOK = true;
+      } else if (hvacComponent.optionalCast<CoilSystemCoolingDXHeatExchangerAssisted>()) {
+        isTypeOK = true;
+      }
```

Comment:
Similarly to above, this is is fine. Technically in OS SDK right now this is uncessary, but if we were to wrap "Coil:Cooling:DX:SingleSpeed:ThermalStorage" then the protection would be needed, so leave it.